chain/boltdb/store.go

```diff
@@ -33,9 +36,27 @@ const BoltFileName = "drand.db"
 // BoltStoreOpenPerm is the permission we will use to read bolt store file from disk
 const BoltStoreOpenPerm = 0660
 
+type newDBFormat bool
+
+var useNewDBFormat newDBFormat = true
+
+func IsATest(ctx context.Context) context.Context {
+	return context.WithValue(ctx, useNewDBFormat, useNewDBFormat)
+}
+
+func isThisATest(ctx context.Context) bool {
+	_, ok := ctx.Value(useNewDBFormat).(newDBFormat)
+	return ok
+}
```

Comment:
I'll address this and https://github.com/drand/drand/pull/1133#discussion_r1089088001 here.
The reason we need these couple of functions is strictly for tests to also use this storage engine.
Otherwise, we wouldn't be able to test it since the default is to use the new one.
Since they are related to the storage itself, then it's OK for them to sit here.

If we go with the suggestion to use this via an environment variable, then we can remove it altogether.

Reply:
Let's keep it as is for now